### ISSUE TITLE
Is it possible for porus to support external crate?

### ISSUE THREAD

**Issue by seekstar** [bhuztez/porus#26]
It seems that porus only supports single rust file that uses libraries provided by porus. But I think it is somewhat inconvenient if the user wants to use an external crate that does not depend on `std`, which should be compatible with porus.

**Comment by bhuztez** [comment]
It is better to make porus a normal crate. Now, unfortuanately there is a crate called porus on crates.io. We have to come up with a new name. However, `cargo` is not as flexible as `gcc` + `make`. Just take a look  `c.py`, it is full of workaround. you could see that it will break as soon as the implementation of cargo changed.

External crate might not be as useful as you expected. They would easily exceed the code size limit of an online judge. We may have to find the implementation with smallest code size. This is why I am working on converting llvm bitcode to C, because generated asm is too big for a BTreeMap.

**Comment by seekstar** [comment]
I don't think we have to publish it on crates.io. I think it's fairly enough to host the crate on GitHub. For example, I host [splay-safe-rs](https://github.com/seekstar/splay-safe-rs) on GitHub and it works fine. If you think it's better to publish on crates.io, I propose the name `porus-oj`.

The code size limit is indeed a problem. Looking forward to your llvm bitcode solution :)
